packages/code-du-travail-frontend/server/server.js

```diff
@@ -24,27 +25,40 @@ const disallowRobots = (_, res, next) => {
   next();
 };
 
-const robotTxtHandler = (req, res) => {
-  const robotsProd = [
-    "User-agent: *",
-    "Disallow: /assets/",
-    "Disallow: /images/",
-    "",
-    `Sitemap: ${FRONTEND_HOST}/sitemap.xml`
-  ].join("\n");
-  const robotsDev = ["User-agent: *", "Disallow: /"].join("\n");
+const robotsProd = [
+  "User-agent: *",
+  "Disallow: /assets/",
+  "Disallow: /images/",
+  "",
+  `Sitemap: ${FRONTEND_HOST}/sitemap.xml`
+].join("\n");
+
+const robotsDev = ["User-agent: *", "Disallow: /"].join("\n");
 
+const robotTxtHandler = (req, res) => {
   res
     .status(200)
     .set("Content-Type", "text/plain")
     .send(LIVEPROD ? robotsProd : robotsDev);
 };
 
+const redirectHostname = (req, res, next) => {
+  const isProdUrl = req.get("Host") === PROD_HOSTNAME;
+  if (!isProdUrl) {
+    return res.redirect(301, `https://${PROD_HOSTNAME}${req.originalUrl}`);
+  }
+  return next();
+};
+
 app.prepare().then(() => {
-  const server = express().get("/robots.txt", robotTxtHandler);
+  const server = express();
+
+  server.get("/robots.txt", robotTxtHandler);
   server.use(expressSitemap(getUrls, FRONTEND_HOST));
```

Comment:
should we use the redirect before other middleware ?

Reply:
yes 👍